addons/udes_stock/models/stock_inventory.py

```diff
@@ -84,12 +104,41 @@ def _is_adjusting_reserved(self):
                 return True
         return False
 
+    def _has_theoretical_quantity_changed(self):
+        """Check if any theoretical quantity in the location has changed.
+        """
+        self.ensure_one()
+        for line in self.line_ids:
+            current_qty = sum(line._get_quants().mapped("quantity"))
+            if (line.theoretical_qty or current_qty) and current_qty != line.theoretical_qty:
+                return True
+        return False
```

Comment:
```suggestion
    def _has_theoretical_quantity_changed(self):
        """Check if any theoretical quantity in the location has changed.
                """
        self.ensure_one()
        return any(sum(line._get_quants().mapped("quantity")) != line.theoretical_qty for line in self.line_ids)
```

Reply:
Not quite as we only want to do this check if either of the theoretical values is > 0 (we dont want to return true if we are adding a new quant). But I can definitely use some of this

Reply:
I think its sufficient != because if it is a new quant both will be 0 , theoritical and current_qty. It still will be false.
First control will be False only if line.theoretical_qty = 0 and current_qty = 0, which even on != control it will be false.

Reply:
True, I think I originally got cufused with the many different types of quantity on inventory lines. I'll change to the suggested

Reply:
Changed